identity-account/src/account/account.rs

```diff
@@ -418,6 +434,54 @@ impl Account {
     Ok(())
   }
 
+  /// Push all unpublished changes for the given identity to the tangle in a single message.
+  pub async fn publish_changes<K: IdentityKey>(&self, key: K) -> Result<()> {
+    let identity: IdentityId = self.try_resolve_id(key).await?;
```

Comment:
I think `publish_changes` also needs a write lock to prevent the potential race condition in `publish`?